samza-kv-rocksdb/src/main/scala/org/apache/samza/storage/kv/RocksDbKeyValueStore.scala

```diff
@@ -236,6 +239,13 @@ class RocksDbKeyValueStore(
     trace("Flushed store: %s" format storeName)
   }
 
+  override def checkpoint(id: String): Optional[Path] = {
+    val checkpoint = Checkpoint.create(db)
+    val checkpointPath = dir.getPath + "-" + id
```

Comment:
What happens if the path already exists? Do snapshots get overwritten or does it create unique checkpoint file name every time checkpoint is created?

Reply:
RocksDB checkpoint dir is required to be unique. Otherwise RocksDB will throw: `org.rocksdb.RocksDBException: Directory exists`

In part 5, id will be created using (current time millis + last 3 digit of nanotime), so the directory will typically be unique.